File: axi_compiler/src/compiler/frontends/cpp/AST.cpp
#include <string>
#include <memory>

enum class AxiComplianceLevel { HighSemantic = 0, LowBareMetal = 1, BoundaryBridge = 2 };

struct AxiASTNode {
    int type;
    AxiComplianceLevel compliance;
    std::string identifier;
    std::string payload;
    bool requires_euler_pool_alloc;
    bool is_unsafe_c_kernel_call;
    std::shared_ptr<AxiASTNode> left_child;
    std::shared_ptr<AxiASTNode> right_child;
};

File: axi_compiler/src/compiler/frontends/rust/AST.rs
#[repr(C)]
pub enum AxiComplianceLevel {
    HighSemantic = 0,
    LowBareMetal = 1,
    BoundaryBridge = 2,
}

#[repr(C)]
pub struct AxiASTNode {
    pub node_type: i32,
    pub compliance: AxiComplianceLevel,
    pub identifier: *mut libc::c_char,
    pub payload: *mut libc::c_char,
    pub requires_euler_pool_alloc: bool,
    pub is_unsafe_c_kernel_call: bool,
    pub left_child: *mut libc::c_void,
    pub right_child: *mut libc::c_void,
}
